superset/assets/javascripts/modules/superset.js

```diff
@@ -212,13 +234,22 @@ const px = function (state) {
         token.find('img.loading').show();
         container.fadeTo(0.5, 0.25);
         container.css('height', this.height());
+        const asyncAnnotations = formData.annotation_layers
+          .map(x => runAnnotationQuery(x, timeout, formData));
         $.ajax({
           url: this.jsonEndpoint(formDataExtra),
           timeout: timeout * 1000,
           success: (queryResponse) => {
             try {
-              vizMap[formData.viz_type](this, queryResponse);
               this.done(queryResponse);
+              // render when all the annotations are available
```

Comment:
Currently the  chart is only drawn on the dashboard when all annotations have been loaded. In a future PR I want display the chart as soon as its data becomes available and add an indicator that shows if not all annotations have been fetched and re-renders when all are available.